nova_plugin/keypair.py

```diff
@@ -63,18 +63,24 @@ def create(nova_client, args, **kwargs):
                                       ctx.node.properties['resource_id'],
                                       private_key_path))
         return
-
-    # Raise error if the file existed and the public key is not provided
-    if pk_exists and not public_key:
+    elif private_key_path and pk_exists:
+        # Raise error if the file existed and the public key is not provided
         raise NonRecoverableError(
             "Can't create keypair - private key path already exists: {0}"
             .format(private_key_path))
+    elif not private_key_path and not public_key:
+        raise NonRecoverableError(
+            'One of the following must be provided: \n'
+            '- A public key, in the public_key under the keypair property. \n'
+            '- A private key path, in the private_key_path property. ('
+            'Not recommended.)'
+        )
 
     keypair = nova_client.keypairs.create(keypair['name'], public_key)
 
     set_openstack_runtime_properties(ctx, keypair, KEYPAIR_OPENSTACK_TYPE)
     # Write to private key if we do not provide public key
-    if not public_key:
+    if private_key_path and public_key:
```

Comment:
if the comment is right so i think it should be "and not public key"

Reply:
good point. chhangedd